addons/sale_project/models/project.py

```diff
@@ -394,21 +396,60 @@ def _get_revenues_items_from_sol(self, domain=None, with_action=True):
                             action_params['res_id'] = sale_order_items.id
                         other_revenues['action'] = action_params
         sequence_per_invoice_type = self._get_profitability_sequence_per_invoice_type()
+        sol_ids = [sol_id for sol_read in sale_line_read_group for sol_id in sol_read['ids']]
         return {
             'data': [{'id': invoice_type, 'sequence': sequence_per_invoice_type[invoice_type], **vals} for invoice_type, vals in revenues_dict.items()],
             'total': {'to_invoice': total_to_invoice, 'invoiced': total_invoiced},
-        }
+        }, sol_ids
```

Comment:
Since it is private, and used only in 1 place, I can change the signature.
> No change in the signature of public methods on model (methods not starting with an underscore), as they are part of the public API used by integrated systems
 - https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean
 
I could avoid doing the sig change, but I would have to redo the same query to recup the `sol_ids`, will be a hit on performance.

Reply:
Okay but the problem is if someone did a custom code in that method, we will break that custom, so I guess we should avoid alter the signature on all methods in our models if it is possible. 😕 🙂